manager/orchestrator/updater.go

```diff
@@ -104,7 +104,7 @@ func (u *Updater) Run(ctx context.Context, service *api.Service, tasks []*api.Ta
 	dirtyTasks := []*api.Task{}
 	for _, t := range tasks {
 		if !reflect.DeepEqual(service.Spec.Task, t.Spec) ||
-			!reflect.DeepEqual(service.Endpoint, t.Endpoint) {
+			!reflect.DeepEqual(service.Spec.Endpoint, t.Endpoint.Spec) {
```

Comment:
Tried this out inside Docker and got a nil pointer dereference here, I believe because `t.Endpoint` is nil. Even though we are populating `t.Endpoint` in the orchestrator, we have to check for `nil` here to account for old state.